test/runtime/fs_full.c

```diff
@@ -0,0 +1,107 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/vfs.h>
+#include <assert.h>
+#include <errno.h>
+
+#define BIGDATA "/bigfile"
+#define NEWFILE "/newfile"
+
+int write_blocks(int fd, int nb)
+{
+    uint8_t buf[512];
+    int b;
+    for (b = 0; b < nb; b++) {
+        uint64_t *p = (uint64_t *)buf;
+        while (p < (uint64_t *)(buf + sizeof(buf)))
+            *p++ = b;
+        if (write(fd, buf, sizeof(buf)) <= 0) {
+            printf("failed writing at block %d err '%s'\n", b, strerror(errno));
+            assert(errno == ENOSPC);
+            break;
+        }
+        if (pread(fd, buf, sizeof(buf), 512 * b) != sizeof(buf)) {
+            printf("failed to read written block %d err '%s'\n", b, strerror(errno));
+            break;
+        }
+        p = (uint64_t *)buf;
+        while (p < (uint64_t *)(buf + sizeof(buf))) {
+            if (*p++ != b) {
+                printf("block %d does not match expected pattern in validation\n", b);
+                break;
```

Comment:
This condition should make the test fail, so either assert() or return a value that is guaranteed to be treated as an error in main()

Reply:
Good catch--I changed it to return the value of the failing block